crates/op-rbuilder/src/integration/integration_test.rs

```diff
@@ -144,68 +150,135 @@ mod tests {
         let latest_block = generator.init().await?;
 
         let provider = ProviderBuilder::<Identity, Identity, Optimism>::default()
-            .on_http("http://localhost:1248".parse()?);
+            .connect_http("http://localhost:1248".parse()?);
 
         let mut base_fee = max(
             latest_block.header.base_fee_per_gas.unwrap(),
             MIN_PROTOCOL_BASE_FEE,
         );
         for _ in 0..10 {
+            // We check queued pool to see if it already has tx we sent earlier
+            // We do this in the beginning, because once we insert valid tx the queued tx would move
+            // to pending pool, because nonce gap would be fixed.
+            let mut queued_hash = None;
+            let mut pool_content = provider.txpool_content().await?;
+            if !pool_content.queued.is_empty() {
```

Comment:
you shouldn't need the empty check here since you already have an assertion for the length of the queued transaction later

Reply:
this assert needed on the iteration zero, when we don't have any transaction in the pool
This will store hash that is inside pending pool in case we are on the 1+ iteration